### PR TITLE
Update system-probe buildimage

### PR DESCRIPTION
**Pull request by leeavital**
### What does this PR do?

See https://github.com/DataDog/datadog-agent/pull/5186/files

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -74,7 +74,7 @@ variables:
   DATADOG_AGENT_BUILDERS: v2276047-a7cebcb
   DATADOG_AGENT_WINBUILDIMAGES: v2276047-a7cebcb
   DATADOG_AGENT_ARMBUILDIMAGES: v2287129-4816dc7
-  DATADOG_AGENT_SYSPROBE_BUILDIMAGES: v2287129-4816dc7
+  DATADOG_AGENT_SYSPROBE_BUILDIMAGES: v2319958-ace4578
```
Comment (leeavital) [inline .gitlab-ci.yml]: This should be replaced with a build off master

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -311,12 +311,12 @@ run_tests_rpm-x64-py3:
...
   stage: source_test
   needs: ["fail_on_non_triggered_tag", "build_libbcc_x64"]
-  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/system-probe_x64:$DATADOG_AGENT_BUILDIMAGES
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/system-probe_x64:$DATADOG_AGENT_SYSPROBE_BUILDIMAGES
   before_script:
     - mkdir -p $CI_PROJECT_DIR/.tmp/binary-ebpf
     - cd $SRC_PATH
     - pip3 install distro -c requirements.txt # needed for some of our invoke tasks until the builders image contain it
-    - inv -e deps --verbose --dep-vendor-only
+    - inv -e deps --verbose --dep-vendor-only --no-checks
```
Comment (kevinconaway) [inline .gitlab-ci.yml]: Why is this needed?

Comment (leeavital) [reply .gitlab-ci.yml]: This task uses an old version of pip and it was causing CI to fail.

A bonus is that without this flag the `dep` task pointlessly downloads the integrations repo, so this addition should actually make our tests run faster.

**Review by kevinconaway — DISMISSED** [review]

**Review by albertvaka — DISMISSED** [review]

**Review by albertvaka — APPROVED** [review]
